src/JSInterop/Microsoft.JSInterop/src/DotNetObjectRefOfT.cs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.JSInterop
+{
+    /// <summary>
+    /// Wraps a JS interop argument, indicating that the value should not be serialized as JSON
+    /// but instead should be passed as a reference.
+    ///
+    /// To avoid leaking memory, the reference must later be disposed by JS code or by .NET code.
+    /// </summary>
+    /// <typeparam name="TValue">The type of the value to wrap.</typeparam>
+    public sealed class DotNetObjectRef<TValue> : IDotNetObjectRef, IDisposable where TValue : class
+    {
+        private long? _trackingId;
+
+        /// <summary>
+        /// This A is for meant for JSON deserialization and should not be used by user code.
+        /// </summary>
+        [Obsolete("This API is meant for JSInterop infrastructure and should not be used by user code.")]
```

Comment:
The empty ctor and TrackingId property are meant for the serializer and are not used by any of our code. I marked them as EditorBrowseableNever and also slapped an Obsolete attribute for good measure. It's not really obsolete, but at least it further deters accidental use.

Reply:
The other alternative would be to write an analyzer for it, which is more work.

Reply:
Is there a risk that using `[Obsolete]` might trigger audit warnings telling us we're supposed to remove obsolete APIs at some point?

It might be sufficient to use `[EditorBrowsable(EditorBrowsableState.Never)]` only, since AFAIK nothing bad can happen if someone ignores us and uses it anyway. They just end up with an instance whose `_trackingId` is fixed at `null` and can never do anything useful, right?

Reply:
> It might be sufficient to use [EditorBrowsable(EditorBrowsableState.Never)] only, since AFAIK nothing bad can happen if someone ignores us and uses it anyway. 

Yup, I suppose. I'll remove the `Obsolete` warnings